ros2-library/src/main/java/us/ihmc/ros2/ROS2NodeBuilder.java

```diff
@@ -0,0 +1,271 @@
+package us.ihmc.ros2;
+
+import us.ihmc.log.LogTools;
+import us.ihmc.pubsub.attributes.ParticipantProfile;
+import us.ihmc.util.PeriodicNonRealtimeThreadSchedulerFactory;
+import us.ihmc.util.PeriodicThreadSchedulerFactory;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.Properties;
+
+/**
+ * A builder to construct {@link ROS2Node}, {@link RealtimeROS2Node}.
+ * <p>
+ * Basic usage:
+ *
+ * <pre>{@code
+ *    ROS2Node node = new ROS2Builder().build("Node");
+ *    RealtimeROS2Node realtimeSHMOnlyNode = new ROS2NodeBuilder().transportSpecial(TransportSpecial.SHARED_MEMORY_ONLY).buildRealtime("SHMNode");
+ * }</pre>
+ * </p>
+ */
+public class ROS2NodeBuilder
+{
+   public enum TransportSpecial
+   {
+      SHARED_MEMORY_ONLY, LOOPBACK_ADDRESS_ONLY, UDPV4_ONLY
+   }
+
+   private static final int UNSET_DOMAIN_ID = -1;
+
+   private int domainId = UNSET_DOMAIN_ID;
+   private String namespace = "/us/ihmc";
+   private boolean useSharedMemory = true;
```

Comment:
Noob question: what's the difference between this field and SHARED_MEMORY_ONLY? Also, should it be true by default?